### PR TITLE
refine LAC computation of avg_speed

### LINKED ISSUE #8287: tiflash metric of resource_group.avg_speed is negitave
## Bug Report
For the current implementation,  avg_speed may be negative because of worker consume too many RU, which is confusing:
image

Please answer these questions before submitting your issue. Thanks!

### PR DESCRIPTION
**Pull request by guo-shaoge**
### What problem does this PR solve?

Issue Number: close #8287

Problem Summary:
1. refine calculation of avg_speed:
    1. before: avg_speed may be negative because of worker consume too many RU, like:
        1. e56c27be-1399-4a78-9ff2-3faab7446918
    2. after: calculate avg_speed every 5 seconds. And `ru_consumption_speed` is the avg speed of last 5 seconds. `ru_consumption_delta` is the ru consumption of last 5 seconds, it will be set to zero after report consumption to GAC every 5 seconds.
        1. image

Others:
1. Add `needNotifyStopTrickleMode` to exit trickle mode timely. Otherwise it's possiable LAC cannot talk to GAC because always got enough token(a.k.a. the fill rate of trickle mode is large enough)
4. Change `ACQUIRE_RU_AMPLIFICATION` from 1.1 to 1.5. Although tidb also use 1.1, but IMO it's a little small.     

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/ResourceControl/TokenBucket.h [hunk 1]
```diff
@@ -83,21 +86,14 @@ class TokenBucket final
...
     std::string toString() const
     {
         FmtBuffer fmt_buf;
-        fmt_buf.fmtAppend(
-            "tokens: {}, fill_rate: {}, capacity: {}, avg_speed_per_sec: {}",
-            tokens,
-            fill_rate,
-            capacity,
-            avg_speed_per_sec);
+        fmt_buf.fmtAppend("tokens: {}, fill_rate: {}, capacity: {}", tokens, fill_rate, capacity);
```
Comment (SeaRise) [inline dbms/src/Flash/ResourceControl/TokenBucket.h:96]:
```suggestion
        return fmt::format("tokens: {}, fill_rate: {}, capacity: {}", tokens, fill_rate, capacity);
```

File: dbms/src/Flash/ResourceControl/LocalAdmissionController.cpp [hunk 2]
```diff
@@ -167,23 +167,41 @@ std::optional<LocalAdmissionController::AcquireTokenInfo> LocalAdmissionControll
     const ResourceGroupPtr & resource_group,
     bool is_periodically_fetch)
 {
-    double token_consumption = resource_group->getAndCleanConsumptionDelta();
+    double token_consumption = 0.0;
     double acquire_tokens = 0.0;
+    const auto now = std::chrono::steady_clock::now();
```
Comment (SeaRise) [inline dbms/src/Flash/ResourceControl/LocalAdmissionController.cpp:172]: can use `using SteadyClock = std::chrono::steady_clock;` like https://github.com/pingcap/tiflash/blob/a55a10fbd6e1d9588e0a1c63de224c76527261c5/dbms/src/Common/MPMCQueue.h#L273 to simplify the code

File: dbms/src/Flash/ResourceControl/LocalAdmissionController.h [hunk 3]
```diff
@@ -326,17 +356,26 @@ class ResourceGroup final : private boost::noncopyable
         return bucket_mode == trickle_mode && tp < stop_trickle_timepoint;
     }
 
-    void collectMetrics() const
+    bool needNotifyStopTrickleMode(const std::chrono::steady_clock::time_point & tp)
```
Comment (SeaRise) [inline dbms/src/Flash/ResourceControl/LocalAdmissionController.h:359]:
```suggestion
    bool overTrickleModeLease(const std::chrono::steady_clock::time_point & tp)
```
just like `inTrickleModeLease`?

Comment (guo-shaoge) [reply dbms/src/Flash/ResourceControl/LocalAdmissionController.h]: changed

File: dbms/src/Flash/ResourceControl/LocalAdmissionController.cpp (hunk 2 above)
Comment (guo-shaoge) [reply dbms/src/Flash/ResourceControl/LocalAdmissionController.cpp]: done

File: dbms/src/Flash/ResourceControl/TokenBucket.h (hunk 1 above)
Comment (guo-shaoge) [reply dbms/src/Flash/ResourceControl/TokenBucket.h]: done

**Review by windtalker — APPROVED** [review]
LGTM

**Review by SeaRise — APPROVED** [review]

**Pull request merged**